hazelcast/src/test/java/com/hazelcast/test/annotation/TestProperties.java

```diff
@@ -0,0 +1,12 @@
+package com.hazelcast.test.annotation;
+
+import java.lang.annotation.Target;
+import java.lang.annotation.Retention;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.RetentionPolicy;
+
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface TestProperties {
+    String maxThreadsCalculatorClass();
```

Comment:
In this case you will have to separatelly implement maxThreads callucaltion for each new test.
In this case you will be able to refer on existent calculator